### ISSUE TITLE
Add support for rule_error? to check if other rules failed.

### ISSUE THREAD

**Issue by graudeejs** [dry-rb/dry-validation#658]
`#rule_error?` can be used to check if there have been errors in current rule. However unlike `#schema_error?`, it can't be used to check if other rule has failed.

Please add support to `#rule_error?` to check if other rules have failed by passing argument similar to `#schema_error?`

## Examples

```ruby
class PersonContract < Dry::Validation::Contract
  schema do
    required(:email).filled(:string)
    required(:name).filled(:string)
  end

  rule(:name) do
    key.failure('expected')
  end

  rule(:email) do
    key.failure('also expected') if rule_error?(:name)
  end
end

PersonContract.new.call(email: "bar", name: 'foo').errors.to_h

## Expected result
# {:name=>["expected"], :email => ["also expected"]}
```

**Comment by adamransom** [comment]
Now that we have https://github.com/dry-rb/dry-validation/pull/656, I currently use `result.error?(:name)` inside of `rule` blocks (which works).

I wonder if it is worth just un-deprecating `error?` and allow that to check for both rule and schema errors for any key? Is there a case where we would just want `rule_error?` (if you depend on the `:name` key in your rule definition, then `error?(:name) works exactly like `rule_error?` would work I think).

**Comment by solnic** [comment]
> Is there a case where we would just want rule_error?

@adamransom yes, that's a legit use case. In fact, I have a couple of such cases in my current project at work 🙂

**Comment by adamransom** [comment]
Fair enough! I still think it's worth allowing `error?` inside of `rule` blocks to check for the presence of either schema or rule errors, but that's probably for another issue 😉

**Comment by solnic** [comment]
> Fair enough! I still think it's worth allowing error? inside of rule blocks to check for the presence of either schema or rule errors, but that's probably for another issu

@adamransom yes this makes sense as well!

**Comment by sirfilip** [comment]
i can pick this one up

**Comment by solnic** [comment]
Closed via #673

**Comment by solnic** [comment]
@sirfilip thanks but actually @moofkit addressed it in #673 which I believe was opened before your comment here 🙂

**Comment by solnic** [comment]
This was done via #673
